File: ROS_YuXiang/test5.2_tf_python/src/demo_python_tf/demo_python_tf/static_tf_broadcaster.py
import math
import rclpy
from rclpy.node import Node
from tf2_ros import StaticTransformBroadcaster
from geometry_msgs.msg import TransformStamped
from tf_transformations import quaternion_from_euler


class StaticTFBroadcaster(Node):
    def __init__(self):
        super().__init__('static_tf_broadcaster')
        self.static_broadcaster_ = StaticTransformBroadcaster(self)
        self.publish_static_tf()

    def publish_static_tf(self):
        static_transform = TransformStamped()
        static_transform.header.stamp = self.get_clock().now().to_msg()
        static_transform.header.frame_id = 'base_link'
        static_transform.child_frame_id = 'camera_link'
        static_transform.transform.translation.x = 0.5
        static_transform.transform.translation.y = 0.3
        static_transform.transform.translation.z = 0.6
        # 旋转
        rotation_quat = quaternion_from_euler(math.radians(180), 0, 0)
        static_transform.transform.rotation.x = rotation_quat[0]
        static_transform.transform.rotation.y = rotation_quat[1]
        static_transform.transform.rotation.z = rotation_quat[2]
        static_transform.transform.rotation.w = rotation_quat[3]
        # 发布
        self.static_broadcaster_.sendTransform(static_transform)
        self.get_logger().info(f'Publish static tf {static_transform}')


def main():
    rclpy.init()
    node = StaticTFBroadcaster()
    rclpy.spin(node)
    rclpy.shutdown()

File: ROS_YuXiang/test5.2_tf_python/src/demo_cpp_tf/src/static_tf_broadcaster.cpp
#include <memory>
#include "geometry_msgs/msg/transform_stamped.hpp"
#include "rclcpp/rclcpp.hpp"
#include "tf2/LinearMath/Quaternion.h"
#include "tf2_geometry_msgs/tf2_geometry_msgs.hpp"
#include "tf2_ros/static_transform_broadcaster.h"

class StaticTFBroadcaster : public rclcpp::Node
{
public:
    StaticTFBroadcaster() : Node("static_tf_broadcaster")
    {
        broadcaster_ = std::make_shared<tf2_ros::StaticTransformBroadcaster>(this);
        this->publish_tf();
    }

    void publish_tf()
    {
        geometry_msgs::msg::TransformStamped t;
        t.header.stamp = this->get_clock()->now();
        t.header.frame_id = "map";
        t.child_frame_id = "target_point";
        t.transform.translation.x = 5.0;
        t.transform.translation.y = 3.0;
        t.transform.translation.z = 0.0;
        tf2::Quaternion q;
        q.setRPY(0, 0, 60 * M_PI / 180);
        t.transform.rotation = tf2::toMsg(q);
        broadcaster_->sendTransform(t);
    }

private:
    std::shared_ptr<tf2_ros::StaticTransformBroadcaster> broadcaster_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<StaticTFBroadcaster>());
    rclcpp::shutdown();
    return 0;
}